cdap-api/src/main/java/co/cask/cdap/api/common/RuntimeArguments.java

```diff
@@ -74,37 +76,85 @@ private RuntimeArguments() {
   }
 
   /**
-   * Extract all arguments for a given scope.
+   * Extracts the scope from the argument as identified by its prefix.
+   *
+   * 1. Argument can be prefixed by "<scopetype>.<name>.". e.g. mapreduce.myMapReduce.read.timeout=30. In this case
+   * only MapReduce program named 'myMapReduce' will receive the argument read.timeout=30.
+   * 2. Argument can be prefixed by "<scopetype>.*.". e.g. mapreduce.*.read.timeout=30. In this case all the underlying
+   * MapReduce programs will receive the argument read.timeout=30.
+   * 3. If the argument is not prefixed with any scope, it is considered global and is passed further without any
+   * changes. e.g. read.timeout=30
+   *
+   * Arguments scoped by DATASET are only filtered while inside the dataset scope. They are passed from other
```

Comment:
Thanks @anew for the review! 
`But it is better to include some irrelevant ones than to exclude some relevant ones.` completely agree! I will update the PR and send out for review soon. It should be way smaller now :)